sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +380,485 @@ def transform(self, X):
+
+        # initial imputation
+        self.initial_imputer_ = None
+        X, X_filled, mask_missing_values = self._initial_imputation(X)
+        n_samples, n_features = X.shape
+
+        # perform imputations
+        total_rounds = self.n_burn_in + self.n_imputations
+        results_list = []
+        self.trained_estimator_triplets_ = []
+        if self.verbose:
+            print("[MICE] Completing matrix with shape %s" % (X.shape,))
+            start_t = time()
+        for iter in range(total_rounds):
+            # order in which to impute
+            ordered_idx = self._get_ordered_idx(mask_missing_values)
+
+            # abs_correlation matrix is used to choose a subset of other
+            # features to impute from
+            abs_corr_mat = self._get_abs_correlation_matrix(X_filled)
+
+            # Fill in each feature in the order of ordered_idx
+            for feat_idx in ordered_idx:
+                neighbor_feat_idx = self._get_neighbor_feat_idx(n_features,
+                                                                feat_idx,
+                                                                abs_corr_mat)
+                X_filled, estimator = self._impute_one_feature(
+                    X_filled, mask_missing_values, feat_idx, neighbor_feat_idx,
+                    estimator=None)
+                estimator_triplet = (feat_idx, neighbor_feat_idx, estimator)
+                self.trained_estimator_triplets_.append(estimator_triplet)
+
+            if iter >= self.n_burn_in:
+                results_list.append(X_filled[mask_missing_values])
+            if self.verbose:
+                print('[MICE] Ending imputation round round '
+                      '%d/%d, elapsed time %0.2f'
+                      % (iter + 1, total_rounds, time() - start_t))
+
+        if len(results_list) > 0:
+            X[mask_missing_values] = np.array(results_list).mean(axis=0)
+        else:
+            X[mask_missing_values] = X_filled[mask_missing_values]
+
+        return X
+
+    def transform(self, X):
+        """Imputes all missing values in X.
+
+        Parameters
+        ----------
+        X : array-like}, shape = [n_samples, n_features]
+            The input data to complete.
+
+        Returns
+        -------
+        X : array-like, shape (n_samples, n_features)
+            The imputed input data.
+        """
+        check_is_fitted(self, 'initial_imputer_')
+
+        # initial imputation
+        X, X_filled, mask_missing_values = self._initial_imputation(X)
+
+        # perform imputations
+        total_rounds = self.n_burn_in + self.n_imputations
+        total_iterations = len(self.trained_estimator_triplets_)
+        imputations_per_round = total_iterations / total_rounds
+        iter = 0
+        results_list = []
+        if self.verbose:
+            print("[MICE] Completing matrix with shape %s" % (X.shape,))
+            start_t = time()
+        for i, estimator_triplet in enumerate(
```

Comment:
Can you rename `i` by `it`

Reply:
I already renamed `iter` by `it`. `i` has a different functionality. It tells me how many steps I've gone through the for loop. I'll try to rename things to clear it up.